spec.html

```diff
@@ -16079,6 +15984,40 @@
           1. Return ? ResolveBinding(_bindingId_).
         </emu-alg>
       </emu-clause>
+
+      <!-- es6num="9.1.11" -->
+      <emu-clause id="sec-enumerate-object-properties" aoid="EnumerateObjectProperties">
+        <h1>EnumerateObjectProperties (_O_)</h1>
+        <p>When the abstract operation EnumerateObjectProperties is called with argument _O_, the following steps are taken:</p>
+        <emu-alg>
+          1. Assert: Type(_O_) is Object.
+          1. Return an Iterator object (<emu-xref href="#sec-iterator-interface"></emu-xref>) whose `next` method iterates over all the String-valued keys of enumerable properties of _O_. The Iterator object must inherit from %IteratorPrototype% (<emu-xref href="#sec-%iteratorprototype%-object"></emu-xref>). The mechanics and order of enumerating the properties is not specified but must conform to the rules specified below.
+        </emu-alg>
+        <p>The iterator's `next` method processes object properties to determine whether the property key should be returned as an iterator value. Returned property keys do not include keys that are Symbols. Properties of the target object may be deleted during enumeration. A property that is deleted before it is processed by the iterator's `next` method is ignored. If new properties are added to the target object during enumeration, the newly added properties are not guaranteed to be processed in the active enumeration. A property name will be returned by the iterator's `next` method at most once in any enumeration.</p>
+        <p>Enumerating the properties of the target object includes enumerating properties of its prototype, and the prototype of the prototype, and so on, recursively; but a property of a prototype is not processed if it has the same name as a property that has already been processed by the iterator's `next` method. The values of [[Enumerable]] attributes are not considered when determining if a property of a prototype object has already been processed. The enumerable property names of prototype objects must be obtained as if by invoking EnumerateObjectProperties passing the prototype object as the argument. EnumerateObjectProperties must obtain the own property keys of the target object as if by calling its [[OwnPropertyKeys]] internal method. Property attributes of the target object must be obtained as if by calling its [[GetOwnProperty]] internal method.</p>
```

Comment:
"as if by calling" was added in https://bugs.ecmascript.org/show_bug.cgi?id=4107, because we need the same set of property names, but we don't want to require to call \[[OwnPropertyKeys]] (or \[[GetOwnProperty]]). From https://github.com/getify/You-Dont-Know-JS/issues/423#issuecomment-97934938:
> "as if" has a specific meaning in standards and is fine to use in normative text. "as if x" means: the observable results will be the same as you would get if you did x but it is ok to get those results some other way.

The last part ("get those results some other way") is clearly not applicable for proxy objects (and we don't want to mislead implementers to do something other than calling \[[OwnPropertyKeys]] on proxies), so it seems wrong to me to leave it in.